SwrveSDK/src/main/java/com/swrve/sdk/messaging/ui/SwrveInAppMessageActivity.java

```diff
@@ -0,0 +1,155 @@
+import android.content.Intent;
+import android.content.pm.ActivityInfo;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.Window;
+import android.view.WindowManager;
+
+import com.swrve.sdk.SwrveBase;
+import com.swrve.sdk.SwrveHelper;
+import com.swrve.sdk.SwrveLogger;
+import com.swrve.sdk.SwrveSDK;
+import com.swrve.sdk.config.SwrveConfigBase;
+import com.swrve.sdk.messaging.SwrveButton;
+import com.swrve.sdk.messaging.SwrveMessage;
+import com.swrve.sdk.messaging.SwrveMessageFormat;
+import com.swrve.sdk.messaging.SwrveOrientation;
+import com.swrve.sdk.messaging.view.SwrveMessageView;
+import com.swrve.sdk.messaging.view.SwrveMessageViewBuildException;
+
+public class SwrveInAppMessageActivity extends Activity {
+
+    protected static final String LOG_TAG = "SwrveMessagingSDK";
+    public static final String MESSAGE_ID_KEY = "message_id";
+
+    private SwrveBase sdk;
+    private SwrveMessage message;
+    private boolean hideToolbar = false;
+    private int minSampleSize;
+    private int defaultBackgroundColor;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        sdk = (SwrveBase)SwrveSDK.getInstance();
+        if (sdk == null) {
+            finish();
+            return;
+        }
+        Intent intent = getIntent();
+        if (intent != null) {
+            Bundle extras = intent.getExtras();
+            if (extras != null) {
+                int messageId = extras.getInt(MESSAGE_ID_KEY);
+                message = sdk.getMessageForId(messageId);
+
+                SwrveConfigBase config = sdk.getConfig();
+                this.hideToolbar = config.isHideToolbar();
+                this.minSampleSize = config.getMinSampleSize();
+                this.defaultBackgroundColor = config.getDefaultBackgroundColor();
+            }
+        }
+
+        if (message == null) {
+            finish();
+            return;
+        }
+        // Choose the current orientation. If it is not possible,
+        // pick the first one and set the requested orientation.
+        SwrveOrientation deviceOrientation = getDeviceOrientation();
+        SwrveMessageFormat format = message.getFormat(deviceOrientation);
+        if (format == null) {
+            format = message.getFormats().get(0);
+            if (format.getOrientation() != deviceOrientation) {
+                if (format.getOrientation() == SwrveOrientation.Landscape) {
+                    setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+                } else {
+                    setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+                }
+            }
+        }
+
+        if (hideToolbar) {
+            // Remove the status bar from the activity
+            getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
+            requestWindowFeature(Window.FEATURE_NO_TITLE);
```

Comment:
The thing about doing it this way is that the title gets removed after the activity is created, thus the titlebar flashes on screen for a split second. The flash can be avoided by configuring the titlebar in the theme rather than programmatically.

If the default is to hide the toolbar, then the default theme on the activity should hide the toolbar, and then programmatically change the theme to one that has a titlebar.